### PR TITLE
Postsharp readiness

### PR DESCRIPTION
**Pull request by gfraiteur**
The changes expose `IEnumerable<IMetadataReference> References { get; }` to `IAfterCompileContext` and `IBeforeCompileContext`. These changes make it possible to integrate PostSharp as an `ICompileModule` because it must know the location of all dependencies.

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@gfraiteur__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
    </span>

TTYL, MSOTBOT;

File: src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IAfterCompileContext.cs [hunk 1]
```diff
@@ -21,5 +22,7 @@ public interface IAfterCompileContext
         Stream XmlDocStream { get; set; }
 
         IList<Diagnostic> Diagnostics { get; }
+
+        IEnumerable<IMetadataReference> References { get; }
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IAfterCompileContext.cs]: Make it `IList<IMetadataReference>` like everything else on the context

File: src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IBeforeCompileContext.cs [hunk 2]
```diff
@@ -16,5 +17,7 @@ public interface IBeforeCompileContext
         IList<ResourceDescription> Resources { get; }
 
         IList<Diagnostic> Diagnostics { get; }
+
+        IEnumerable<IMetadataReference> References { get; }
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IBeforeCompileContext.cs]: Same comment as above

File: src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IAfterCompileContext.cs (hunk 1 above)
Comment (gfraiteur) [inline src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IAfterCompileContext.cs]: Do you want it immutable like in Roslyn? Not sure of what's your convention in this project.

Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IAfterCompileContext.cs]: Nope, just follow the convention on the current interface for now. We're going to make other changes later.

**Comment by msotclas** [comment]
__@gfraiteur__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSOTBOT;

**Comment by gfraiteur** [comment]
@davidfowl  The changes you asked are ready.

**Comment by troydai** [comment]
Thanks for the contribution, @gfraiteur. I just merged your change.

**Pull request merged**